### PR TITLE
fix: Avoids error when removing project api key assignment for deleted project

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): [INTMDB-1297](https://jira.mongodb.org/browse/INTMDB-1297)

Enables deleting a project and is project_assignment in project_api_key resource without encountering errors. This is only one part of [HELP-52313](https://jira.mongodb.org/browse/HELP-52313.).

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_project_api_key.go [hunk 1]
```diff
@@ -396,30 +399,30 @@ func newProjectAssignment(ctx context.Context, conn *matlas.Client, apiKeyID str
 	return results, nil
 }
 
-func getStateProjectAssignmentAPIKeys(d *schema.ResourceData) (newAPIKeys, changedAPIKeys, removedAPIKeys []any) {
```
Comment (AgustinBettati) [inline mongodbatlas/resource_mongodbatlas_project_api_key.go:399]: changes in this function are only renaming variables

File: mongodbatlas/resource_mongodbatlas_project_api_key.go [hunk 2]
```diff
@@ -209,17 +209,20 @@ func resourceMongoDBAtlasProjectAPIKeyUpdate(ctx context.Context, d *schema.Reso
...
 		}
 
-		// Removing api_keys from the project
-		for _, apiKey := range removedAPIKeys {
+		// Removing projects assignments
+		for _, apiKey := range removedAssignments {
 			projectID := apiKey.(map[string]any)["project_id"].(string)
 			_, err := conn.ProjectAPIKeys.Unassign(ctx, projectID, apiKeyID)
+			if err != nil && strings.Contains(err.Error(), "GROUP_NOT_FOUND") {
+				continue // allows removing assignment for a project that has been deleted
```
Comment (lantoli) [inline mongodbatlas/resource_mongodbatlas_project_api_key.go:217]: i would normally avoid continue in this situation, although your solution is probably more clear than:

if err != nil { 
    if !strings.Contains(err.Error(), "GROUP_NOT_FOUND") { // allows removing assignment for a project that has been deleted
				return diag.Errorf("error removing api_key(%s) from the project(%s): %s", apiKeyID, projectID, err)
}
}

**Review by lantoli — APPROVED** [review]
knitpit for changelog in PR title:

fix: avoid error when removing project api key...  =>

fix: Avoids error when removing project api key...

**Review by oarbusi — APPROVED** [review]

File: mongodbatlas/resource_mongodbatlas_project_api_key.go (hunk 2 above)
Comment (AgustinBettati) [reply mongodbatlas/resource_mongodbatlas_project_api_key.go:217]: Yeah, went for this approach to make this case explicit and easy to read

**Pull request merged**